include/mbedtls/pk.h

```diff
@@ -232,12 +232,20 @@ typedef struct mbedtls_pk_debug_item {
  */
 typedef struct mbedtls_pk_info_t mbedtls_pk_info_t;
 
+#define MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN \
+    PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE(PSA_VENDOR_ECC_MAX_CURVE_BITS)
 /**
  * \brief           Public key container
  */
 typedef struct mbedtls_pk_context {
     const mbedtls_pk_info_t *MBEDTLS_PRIVATE(pk_info);    /**< Public key information         */
     void *MBEDTLS_PRIVATE(pk_ctx);                        /**< Underlying public key context  */
+#if defined(MBEDTLS_ECP_C) && defined(MBEDTLS_USE_PSA_CRYPTO)
+    uint8_t MBEDTLS_PRIVATE(pk_raw)[MBEDTLS_PK_MAX_EC_PUBKEY_RAW_LEN];  /**< Raw public key   */
```

Comment:
Yes, even though the 1st option looks more elegant, I suspect that it includes more changes than the 2nd one. Indeed albeit the function calls remain the same, then you'll still have to handle the new structure instead of the previous `ecp_keypair`. 
Of course having the `USE_PSA` solution to work in place of the legacy one instead of together would simplify things as in one case we might have the `ecp_keypair` while in the other one a new structure containing the raw public key and (in future) also the private key ID.